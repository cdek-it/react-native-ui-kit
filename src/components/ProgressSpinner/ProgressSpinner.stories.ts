import type { Meta, StoryObj } from '@storybook/react'

import { ProgressSpinner } from './ProgressSpinner'

const meta: Meta<typeof ProgressSpinner> = {
  title: 'ProgressSpinner',
  component: ProgressSpinner,
  args: { size: 'md', fill: 'primary' },
  argTypes: {
    fill: { control: 'radio', options: ['primary', 'white'] },
    size: { control: 'radio', options: ['xl', 'lg', 'md', 'sm'] },
  },
}

export default meta

type Story = StoryObj<typeof ProgressSpinner>

const ProgressSpinnerStory: Story = {}

export { ProgressSpinnerStory as ProgressSpinner }
