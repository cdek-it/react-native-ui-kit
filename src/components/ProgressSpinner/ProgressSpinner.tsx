import { memo, useEffect, useMemo } from 'react'
import Animated, {
  Easing,
  interpolate,
  useAnimatedProps,
  useAnimatedStyle,
  useSharedValue,
  withRepeat,
  withTiming,
} from 'react-native-reanimated'
import Svg, { Circle } from 'react-native-svg'

import { makeStyles } from '../../utils/makeStyles'

// eslint-disable-next-line import-x/no-deprecated
const AnimatedSvg = Animated.createAnimatedComponent(Svg)
// eslint-disable-next-line import-x/no-deprecated
const AnimatedCircle = Animated.createAnimatedComponent(Circle)

const STROKE_WIDTH = 2

export interface ProgressSpinnerProps {
  /**
   * Выбор размера компонента
   * @default 'md'
   */
  size?: 'xl' | 'lg' | 'md' | 'sm'

  /**
   * Выбор цвета компонента
   * @default 'primary'
   */
  fill?: 'primary' | 'white'
}

/**
 * Используется для отображения состояний ожидания в интерфейсе
 * @see https://www.figma.com/design/4TYeki0MDLhfPGJstbIicf/UI-kit-PrimeFace-(DS)?node-id=1219-3086
 */
export const ProgressSpinner = memo<ProgressSpinnerProps>(
  ({ size = 'md', fill = 'primary' }) => {
    const styles = useStyles()
    const circleAnimation = useSharedValue(0)
    const containerAnimation = useSharedValue(0)

    const sizeInDp = useMemo(() => {
      switch (size) {
        case 'xl':
          return 56

        case 'lg':
          return 42

        case 'md':
          return 28

        case 'sm':
          return 14
      }
    }, [size])

    const color = useMemo(() => {
      switch (fill) {
        case 'primary':
          return styles.primary.color

        case 'white':
          return styles.white.color
      }
    }, [fill, styles.primary.color, styles.white.color])

    const center = useMemo(() => sizeInDp / 2, [sizeInDp])
    const radius = useMemo(() => center - STROKE_WIDTH / 2, [center])
    const circleLength = useMemo(() => 2 * Math.PI * radius, [radius])

    const containerAnimatedStyle = useAnimatedStyle(() => ({
      transform: [{ rotate: `${containerAnimation.value * 360}deg` }],
    }))

    const circleAnimatedProps = useAnimatedProps(() => ({
      strokeDasharray: [
        interpolate(
          circleAnimation.value,
          [0, 0.5, 1],
          [1, 0.7 * circleLength, circleLength]
        ),
        circleLength,
      ],
      strokeDashoffset: interpolate(
        circleAnimation.value,
        [0, 0.5, 1],
        [0, -(0.3 * circleLength), -circleLength + 2]
      ),
    }))

    useEffect(() => {
      circleAnimation.value = withRepeat(
        withTiming(1, { duration: 1500, easing: Easing.inOut(Easing.ease) }),
        -1,
        false
      )
    }, [circleAnimation])

    useEffect(() => {
      containerAnimation.value = withRepeat(
        withTiming(1, { duration: 2000, easing: Easing.linear }),
        -1
      )
    }, [containerAnimation])

    return (
      <AnimatedSvg
        style={[containerAnimatedStyle, { width: sizeInDp, height: sizeInDp }]}
        testID={TestId.ProgressSpinner}
      >
        <AnimatedCircle
          animatedProps={circleAnimatedProps}
          cx={center}
          cy={center}
          fill='none'
          r={radius}
          stroke={color}
          strokeLinecap='round'
          strokeWidth={STROKE_WIDTH}
        />
      </AnimatedSvg>
    )
  }
)

const useStyles = makeStyles(({ theme, global }) => ({
  primary: { color: theme.General.primaryColor },
  white: { color: global.Neutrals.White['white-100'] },
}))

enum TestId {
  ProgressSpinner = 'ProgressSpinner',
}
