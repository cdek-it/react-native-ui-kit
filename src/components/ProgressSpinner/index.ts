export { ProgressSpinner, type ProgressSpinnerProps } from './ProgressSpinner'
