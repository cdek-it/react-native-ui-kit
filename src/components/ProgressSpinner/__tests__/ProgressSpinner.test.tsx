import { render } from '@testing-library/react-native'

import { ProgressSpinner, type ProgressSpinnerProps } from '../ProgressSpinner'

describe('ProgressSpinner', () => {
  const progressSpinnerSnapshotCases =
    generatePropsCombinations<ProgressSpinnerProps>({
      size: ['xl', 'lg', 'md', 'sm', undefined],
      fill: ['primary', 'white', undefined],
    })

  test.each(progressSpinnerSnapshotCases)('%p', (props) => {
    const { toJSON } = render(<ProgressSpinner {...props} />)

    expect(toJSON()).toMatchSnapshot()
  })
})
