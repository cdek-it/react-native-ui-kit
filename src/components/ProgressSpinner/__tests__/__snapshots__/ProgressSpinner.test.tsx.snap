// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressSpinner {"fill": "primary", "size": "lg"} 1`] = `
<RNSVGSvgView
  bbHeight={42}
  bbWidth={42}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 42,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 42,
      },
      {
        "flex": 0,
        "height": 42,
        "width": 42,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={21}
      cy={21}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={20}
      stroke={
        {
          "payload": 4279797504,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          125.66370614359172,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`ProgressSpinner {"fill": "primary", "size": "md"} 1`] = `
<RNSVGSvgView
  bbHeight={28}
  bbWidth={28}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 28,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 28,
      },
      {
        "flex": 0,
        "height": 28,
        "width": 28,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={14}
      cy={14}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={13}
      stroke={
        {
          "payload": 4279797504,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          81.68140899333463,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`ProgressSpinner {"fill": "primary", "size": "sm"} 1`] = `
<RNSVGSvgView
  bbHeight={14}
  bbWidth={14}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 14,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 14,
      },
      {
        "flex": 0,
        "height": 14,
        "width": 14,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={7}
      cy={7}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={6}
      stroke={
        {
          "payload": 4279797504,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          37.69911184307752,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`ProgressSpinner {"fill": "primary", "size": "xl"} 1`] = `
<RNSVGSvgView
  bbHeight={56}
  bbWidth={56}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 56,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 56,
      },
      {
        "flex": 0,
        "height": 56,
        "width": 56,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={28}
      cy={28}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={27}
      stroke={
        {
          "payload": 4279797504,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          169.64600329384882,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`ProgressSpinner {"fill": "primary", "size": undefined} 1`] = `
<RNSVGSvgView
  bbHeight={28}
  bbWidth={28}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 28,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 28,
      },
      {
        "flex": 0,
        "height": 28,
        "width": 28,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={14}
      cy={14}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={13}
      stroke={
        {
          "payload": 4279797504,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          81.68140899333463,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`ProgressSpinner {"fill": "white", "size": "lg"} 1`] = `
<RNSVGSvgView
  bbHeight={42}
  bbWidth={42}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 42,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 42,
      },
      {
        "flex": 0,
        "height": 42,
        "width": 42,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={21}
      cy={21}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={20}
      stroke={
        {
          "payload": 4294967295,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          125.66370614359172,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`ProgressSpinner {"fill": "white", "size": "md"} 1`] = `
<RNSVGSvgView
  bbHeight={28}
  bbWidth={28}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 28,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 28,
      },
      {
        "flex": 0,
        "height": 28,
        "width": 28,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={14}
      cy={14}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={13}
      stroke={
        {
          "payload": 4294967295,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          81.68140899333463,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`ProgressSpinner {"fill": "white", "size": "sm"} 1`] = `
<RNSVGSvgView
  bbHeight={14}
  bbWidth={14}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 14,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 14,
      },
      {
        "flex": 0,
        "height": 14,
        "width": 14,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={7}
      cy={7}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={6}
      stroke={
        {
          "payload": 4294967295,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          37.69911184307752,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`ProgressSpinner {"fill": "white", "size": "xl"} 1`] = `
<RNSVGSvgView
  bbHeight={56}
  bbWidth={56}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 56,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 56,
      },
      {
        "flex": 0,
        "height": 56,
        "width": 56,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={28}
      cy={28}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={27}
      stroke={
        {
          "payload": 4294967295,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          169.64600329384882,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`ProgressSpinner {"fill": "white", "size": undefined} 1`] = `
<RNSVGSvgView
  bbHeight={28}
  bbWidth={28}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 28,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 28,
      },
      {
        "flex": 0,
        "height": 28,
        "width": 28,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={14}
      cy={14}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={13}
      stroke={
        {
          "payload": 4294967295,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          81.68140899333463,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`ProgressSpinner {"fill": undefined, "size": "lg"} 1`] = `
<RNSVGSvgView
  bbHeight={42}
  bbWidth={42}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 42,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 42,
      },
      {
        "flex": 0,
        "height": 42,
        "width": 42,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={21}
      cy={21}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={20}
      stroke={
        {
          "payload": 4279797504,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          125.66370614359172,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`ProgressSpinner {"fill": undefined, "size": "md"} 1`] = `
<RNSVGSvgView
  bbHeight={28}
  bbWidth={28}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 28,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 28,
      },
      {
        "flex": 0,
        "height": 28,
        "width": 28,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={14}
      cy={14}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={13}
      stroke={
        {
          "payload": 4279797504,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          81.68140899333463,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`ProgressSpinner {"fill": undefined, "size": "sm"} 1`] = `
<RNSVGSvgView
  bbHeight={14}
  bbWidth={14}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 14,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 14,
      },
      {
        "flex": 0,
        "height": 14,
        "width": 14,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={7}
      cy={7}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={6}
      stroke={
        {
          "payload": 4279797504,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          37.69911184307752,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`ProgressSpinner {"fill": undefined, "size": "xl"} 1`] = `
<RNSVGSvgView
  bbHeight={56}
  bbWidth={56}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 56,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 56,
      },
      {
        "flex": 0,
        "height": 56,
        "width": 56,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={28}
      cy={28}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={27}
      stroke={
        {
          "payload": 4279797504,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          169.64600329384882,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`ProgressSpinner {"fill": undefined, "size": undefined} 1`] = `
<RNSVGSvgView
  bbHeight={28}
  bbWidth={28}
  collapsable={false}
  focusable={false}
  jestAnimatedStyle={
    {
      "value": {
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
      },
    }
  }
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "height": 28,
        "transform": [
          {
            "rotate": "0deg",
          },
        ],
        "width": 28,
      },
      {
        "flex": 0,
        "height": 28,
        "width": 28,
      },
    ]
  }
  testID="ProgressSpinner"
  transform={
    [
      {
        "rotate": "0deg",
      },
    ]
  }
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGCircle
      cx={14}
      cy={14}
      fill={null}
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
          "strokeDasharray",
          "strokeDashoffset",
          "strokeLinecap",
        ]
      }
      r={13}
      stroke={
        {
          "payload": 4279797504,
          "type": 0,
        }
      }
      strokeDasharray={
        [
          1,
          81.68140899333463,
        ]
      }
      strokeDashoffset={null}
      strokeLinecap={1}
      strokeWidth={2}
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
